### DISCUSSION TITLE
DDP number of training iterations independent of #GPUs?

### DISCUSSION THREAD

**Discussion by timothybrooks** [Lightning-AI/pytorch-lightning#6337] [DDP / multi-GPU / multi-node]
Hello--

I am using DDP and anywhere from 1-8 GPUs. The PL documentation explains that it will wrap my training DataLoader in a DistributedSampler which gives each process a unique partition of the dataset. In this case, since it is also my understanding that the batch size is per-GPU, I would expect the total number of iterations to decrease proportionally to the number of GPUs. However the number of training batch iterations is constant (in my case ~100k) regardless of how many GPUs I use. It is equal to `dataset_size / batch_size` while I expected it to be `dataset_size / (batch_size * num_gpus)`.

Am I missing something here? Why isn't PL partitioning the dataset such that each process sees a unique subset?

**Comment by mrmotallebi** [comment]
May it be related to my post, that each GPU is loading the whole dataset, thus the number of iterations for your case is constant?

https://github.com/PyTorchLightning/pytorch-lightning/discussions/6350

Edit: Saw that you had already reacted to it. :)

**Reply by timothybrooks** [reply]
Yes I think this is the same underlying issue!

**Reply by mrmotallebi** [reply]
I tried changing the number of GPUs for my own case and the number of iterations changed accordingly.

**Comment by mrmotallebi** [comment]
How are you setting the number of GPUs?  You have to pass it to the Trainer() otherwise it won't have effect.

**Reply by timothybrooks** [reply]
I created the following issue:
https://github.com/PyTorchLightning/pytorch-lightning/issues/6423
